flink-libraries/flink-table/src/test/scala/org/apache/flink/api/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -1101,6 +1101,45 @@ class ScalarFunctionsTest extends ExpressionTestBase {
       "true")
   }
 
+  @Test
+  def testInExpressions(): Unit = {
+    testTableApi(
```

Comment:
You are right, working with POJOs and Case classes does not always give you best performance but limiting the number of supported types results in bad user experience. Types that are supported in general, should also be supported by features. Why can't we use POJOs in SQL? Is `f4 IN (f3, f2, f0)` not possible?